### ISSUE TITLE
Investigate the license in cg_api

### ISSUE THREAD

**Issue by illwieckz** [DaemonEngine/Daemon#281]
See
https://github.com/Unvanquished/Unvanquished/commit/aaf0ab2d627d3dcf76bce8c49ef7986fc4c5f886#diff-2fcc9e06b4d04c78e7336a4722443a0fL244
And
https://github.com/Unvanquished/Unvanquished/commit/8e3469490ed794eec484c89118d2681b819d18ab#diff-7fec21ff46c9d30c133e635643ad925cR76
Reported by @TheDushan

**Comment by illwieckz** [comment]
@Kangz can you look at this? and check what is new and can be relicensed under BSD, and what may be not new and must be kept under the GPL?

**Comment by slipher** [comment]
Apparently only function signatures are copied. I guess it must be OK to copy those, since people often make clones of APIs/programming languages (e.g. MATLAB -> Octave).

**Comment by Kangz** [comment]
TL;DR the current file is ok because all the code and comments from the original file are gone and replaced with new implementations. There was however a small process error that wasn't caught during code review.

Context can be seen in this [file history](https://github.com/Unvanquished/Unvanquished/commits/ca388dc0c6bac110254a4e5a175abbb7963b1f4f?after=ca388dc0c6bac110254a4e5a175abbb7963b1f4f+34&path%5B%5D=src&path%5B%5D=gamelogic&path%5B%5D=cgame&path%5B%5D=cg_api.cpp) between "Active back cgame compilation (no syscalls implemented yet)" and "Brutally implement rocket cgame syscalls".

The issue is that the first comment copies the original while changing the license, and only afterwards the content of the file is changed to a new implementation through 10 commits and two weeks locally on my computer up to "Brutally implement rocket cgame syscalls" at which point cg_api.cpp is completely new. What should have happened is that the original license should have been kept during that process and changed only once the file was completely rewritten.

This wasn't caught during code review because the branch was reviewed as a single unit, and the diff showed a completely new file with the new license, which is ok. So while there was a slight process mistake, the end state of master and tags is fine. No changes or history rewriting are needed.

**Comment by illwieckz** [comment]
Thank you, that answer is very clear.

Based on what you said, here is [the exact diff](https://github.com/Unvanquished/Unvanquished/compare/8e3469490ed794eec484c89118d2681b819d18ab...fc223c80da5d4551f6ef6bd3e1202a802dbbc0f7) between those two commits:

- _“Active back cgame compilation (no syscalls implemented yet)”_
- _“Brutally implement rocket cgame syscalls”_

But that's hard to read because of the large amount of changes.

An easier way to figure out what has been done is to look at the `cg_api.c` and it's copy `cg_api.cpp` with the new header being merged mistakenly too soon, with the same content, this is the first commit of the branch (commit date: 2014-08-31):

[Unvanquished@`8e34694` src/gamelogic/cgame/cg_api.c](https://github.com/Unvanquished/Unvanquished/blob/8e3469490ed794eec484c89118d2681b819d18ab/src/gamelogic/cgame/cg_api.c)
[Unvanquished@`8e34694` src/gamelogic/cgame/cg_api.cpp](https://github.com/Unvanquished/Unvanquished/blob/8e3469490ed794eec484c89118d2681b819d18ab/src/gamelogic/cgame/cg_api.cpp)

But we can see that on the final commit of the branch (commit date: 2014-09-16), everything was rewritten and code is brand new and original:

[Unvanquished@`fc223c8` src/gamelogic/cgame/cg_api.cpp](https://github.com/Unvanquished/Unvanquished/blob/fc223c80da5d4551f6ef6bd3e1202a802dbbc0f7/src/gamelogic/cgame/cg_api.cpp)

Then, if wee look at this file today (commit date: 2019-03-03) the file has changed a lot again:

[Unvanquished@`e3b448a` src/cgame/cg_api.cpp](https://github.com/Unvanquished/Unvanquished/blob/e3b448aafe9f18681dcbaa8636d4625504051372/src/cgame/cg_api.cpp)

So, the header was applied a bit sooner than needed and the issue last for some days in a feature branch before being fixed while the branch was merged as a whole. Everything is fine since 5 years now. There is no need to rewrite the history, this ticket will prevail.

Note: that file does not exist anymore in current tree of Dæmon, it was removed when decoupling engine with game code, the file is tracked in [Unvanquished](https://github.com/Unvanquished/Unvanquished) repository instead.
